configs/projects/_shared-agent-components.rb

```diff
@@ -13,6 +13,14 @@
 # Common components required by all agent branches
 proj.component 'runtime-agent'
 
+matchdata = platform.settings[:ruby_version].match /(\d+)\.\d+\.\d+/
+ruby_major_version = matchdata[1].to_i
+# Ruby 3.2 does not package these two libraries so we need to add them as a component
+if ruby_major_version >= 3
+  proj.component 'libffi'
+  proj.component 'libyaml'
```

Comment:
I really wanted to get this added in the ruby-3.2 component, but I couldn't make it work, so instead added it here with some logic gated to only ruby 3+